evolver/app/routers/hardware.py

```diff
@@ -45,26 +46,32 @@ def get_hardware(hardware_name: str, request: Request):
 
 
 # Start the calibration procedure for the selected hardware and vials
-@router.post("/{hardware_name}/calibrator/start")
+@router.post("/{hardware_name}/calibrator/procedure/start")
 def start_calibration_procedure(
-    hardware_name: str, request: Request, calibration_request: StartCalibrationProcedureRequest
+    hardware_name: str, request: Request, initial_state: TempCalibrationProcedureInitialState | None
 ):
     hardware_instance = get_hardware_instance(request, hardware_name)
     calibrator = hardware_instance.calibrator
     if not calibrator:
         raise HTTPException(status_code=404, detail=f"Calibrator not found for '{hardware_name}'")
 
-    calibrator.initialize_calibration_procedure(
-        selected_hardware=hardware_instance,
-        selected_vials=calibration_request.selected_vials,
-        evolver=request.app.state.evolver,
-    )
+    match hardware_name:
+        case "temp":
+            calibrator.initialize_calibration_procedure(
+                selected_hardware=hardware_instance,
+                initial_state=initial_state,
+                evolver=request.app.state.evolver,
+            )
+        case _:
+            raise HTTPException(
+                status_code=404, detail=f"No initialize_calibration_procedure defined for '{hardware_name}'"
+            )
 
-    return calibrator.state
+    return calibrator.calibration_procedure.get_state()
```

Comment:
refactor, CalibrationProcedure has its own state.

This is eventually persisted to the Calibrator.CalibrationData class. But not until a procedure explicitly dispatches an action to do that.

Reply:
haven't addressed this in this PR.